odata-protocol/11.4 Data Modification.md

```diff
@@ -680,23 +689,45 @@ with the specified media stream if it does not already exist, otherwise the
 media stream of the existing media entity is
 [updated](#UpdateaMediaEntityStream).
 
-Singleton entities can be upserted if they are nullable. Services
-supporting this SHOULD advertise it by annotating the singleton with the
+Singletons and single-valued containment navigation properties can be upserted if they are nullable. Services
+supporting this SHOULD advertise it by an annotation with the
 term `Capabilities.UpdateRestrictions` (nested property `Upsertable`
 with value `true`) defined in [OData-VocCap](#ODataVocCap).
 
 A key property whose value is provided in the request URL SHOULD be omitted from the request body.
 If key properties are provided in the request URL and the request body with different values,
 services MUST either fail the request or ignore the value in the request body.
 
-To ensure that an update request is not treated as an insert, the client
-MAY specify an [`If-Match`](#HeaderIfMatch) header in the update
-request. The service MUST NOT treat an update request containing an
-`If-Match` header as an insert.
-
-A `PUT` or `PATCH` request MUST NOT be treated as an update if an
-[`If-None-Match`](#HeaderIfNoneMatch) header is specified with a value
-of `*`.
+If the URL of an update request falls into case 1, 2, or 3 given above and the request
+- contains an [`If-Match: *`](#HeaderIfMatch) header, the service MUST fail the request
+  if it would newly create the addressed entity in the absence of the header
+- contains an `If-Match` header with an etag as value, the service fails the request
+  instead of newly creating the addressed entity because the [etag](#UseofETagsforAvoidingUpdateConflicts)
+  does not match
+- contains an [`If-None-Match: *`](#HeaderIfNoneMatch) header, the service MUST fail the request
+  if it would update an existing entity in the absence of the header.
+
+In other words, the `If-`(`None-`)`Match` header distinguishes between update and upsert
+in these cases.
+
+If the URL of an update request does not fall into case 1, 2, or 3 given above, it ends
+with a non-containment navigation property, optionally followed by a key predicate.
```

Comment:
In this case, the client may not care whether the operation is an update or upsert, and hence may not want to evaluate both `Updatable` and `Upsertable` in the capabilities annotations.